packages/sanity/src/router/withRouter.tsx

```diff
@@ -3,7 +3,23 @@ import {RouterContextValue} from './types'
 import {useRouter} from './useRouter'
 
 /**
+ * A higher-order component that injects the `router` object from the `useRouter` hook
```

Comment:
I'm all for making it `internal` (I'd also be ok with deprecating it, but since we don't know if there's widespread use beyond our gut feeling I'm unsure if we should)

Reply:
Does it make sense to make it both `@internal` and `@deprecated` so we can remove it in v4?

Reply:
> Does it make sense to make it both @internal and @deprecated so we can remove it in v4?

Yep - not sure if we export things marked as `@internal`, but if we do that makes sense!

Reply:
couldn't find any reference to [stripInternal](https://www.typescriptlang.org/tsconfig#stripInternal), so looks like we still export them. Let's go with  both `@internal` and `@deprecated` then.